eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/DefaultSourcingCondition.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2010-2024. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.eventstore;
+
+import jakarta.annotation.Nonnull;
+
+import java.util.Objects;
+import java.util.OptionalLong;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+/**
+ * The default {@link SourcingCondition} implementation.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+final class DefaultSourcingCondition implements SourcingCondition {
+
+    private final EventCriteria criteria;
+    private final Long start;
```

Comment:
`start` doesn't seem to allow `null`. Shouldn't this be a `long` instead?

Reply:
I think I made this a `Long` because of the static builder methods on the `SourcingCondition`, which look like this:

```java
static SourcingCondition conditionFor(@Nonnull EventCriteria criteria,
                                      Long start,
                                      Long end) {
    return new DefaultSourcingCondition(criteria, start, end);
}
```

Changing the second parameter into a `long` i.o. a `Long` seemed unclear to me.
However, looking at this again, I don't think the `SourcingCondition#end` should return an `OptionalLong` at all. 
A `long` will do just fine.

I will change the `DefaultSourcingCondition` to a `record` with `long` `start` and `end`, as well as adjust the `SourcingCondition` static builder methods to not take `Long` parameters and the `SourcingCondition#end` to not return an `OptionalLong`.